File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        map1 = []
        map2 = []
        for i in s:
            map1.append(s.index(i))
        for j in t:
            map2.append(t.index(j))
        if map1 == map2:
            return True
        return False

File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.size()!=t.size()){
            return false;
        }
        unordered_map<char,char> mpp1;
        unordered_map<char,char> mpp2;
        for(int i=0;i<s.size();i++){
            char s1 = s[i];
            char s2 = t[i];
            if((mpp1.count(s1) && mpp1[s1]!=s2) || (mpp2.count(s2) && mpp2[s2]!=s1)){
                return false;
            }
            mpp1[s1] = s2;
            mpp2[s2] = s1;
        }
        return true;
    }
};